### PR TITLE
[StopWatch] add `elapsed` function

### PR DESCRIPTION
**Pull request by filippobrizzi**
# Description
Add `elapsed` function ot `StopWatch`. The function allows to return the time elapsed from first call to `start()`

### CODE REVIEW

File: modules/utils/include/hephaestus/utils/timing/stop_watch.h [hunk 1]
```diff
@@ -27,13 +27,17 @@ class StopWatch {
...
-  /// - time elapsed from max[most recent start(), most recent lap()] to now.
+  /// - time elapsed from max[most recent start(), most recent lapse()] to now.
   /// - Cast to desired duration.
   /// - Doesn't stop the watch.
   template <typename TargetDurationT = StopWatch::DurationT>
   [[nodiscard]] auto lapse() -> TargetDurationT;
 
-  /// Stop and reset accumlated information.
+  /// \return Elapsed time since the last start() cast to the desired duration.
+  template <typename TargetDurationT = StopWatch::DurationT>
```
Comment (lhruby) [inline modules/utils/include/hephaestus/utils/timing/stop_watch.h:37]: I think the clock template parameter should be common to `StopWatch`, and all `StopWatch` member functions should not have a clock template parameter.

As is, you could define one clock for the StopWatch, and another one for `elapsed`, `stop`, ... This seems like a logic error.

File: modules/utils/include/hephaestus/utils/timing/stop_watch.h [hunk 2]
```diff
@@ -27,13 +27,17 @@ class StopWatch {
   [[nodiscard]] auto stop() -> TargetDurationT;
 
   /// \return Currently running lap time:
```
Comment (lhruby) [inline modules/utils/include/hephaestus/utils/timing/stop_watch.h:29]:
```suggestion
  /// \return Currently running lap time, measured from the last call to lapse. The first lap is measured from the last start timestamp:
```

**Comment by lhruby** [comment]
LGTM. One suggestion, I would add a simply ascii schematic about what each function does to the class header. That will clarify more than any documentation of the individual functions can
```
|     Lapse      |       Lapse     |
|-----------------|------=-----------|
| accumulatedLapsDuration |
``` 
etc. WDYT?

File: modules/utils/include/hephaestus/utils/timing/stop_watch.h (hunk 1 above)
Comment (filippobrizzi) [reply modules/utils/include/hephaestus/utils/timing/stop_watch.h:37]: the clock is indeed the same this is the type of the returned duration you want. in this way you do not have to manually cast between milliseconds and microseconds

**Pull request merged**